com.ibm.wala.util/src/main/java/com/ibm/wala/util/graph/impl/BasicEdgeManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2002 - 2021 IBM Corporation.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ */
+package com.ibm.wala.util.graph.impl;
+
+import com.ibm.wala.util.collections.HashMapFactory;
+import com.ibm.wala.util.collections.MapUtil;
+import com.ibm.wala.util.graph.EdgeManager;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Simple implementation of an {@link com.ibm.wala.util.graph.EdgeManager}. Does not support edge
+ * deletion.
+ */
+public class BasicEdgeManager<T> implements EdgeManager<T> {
+  private final Map<T, Set<T>> preds = HashMapFactory.make();
+
+  private final Map<T, Set<T>> succs = HashMapFactory.make();
+
+  @Override
+  public Iterator<T> getPredNodes(T n) {
+    Set<T> nodePreds = this.preds.get(n);
+    return nodePreds != null ? nodePreds.iterator() : Collections.<T>emptySet().iterator();
```

Comment:
Use `Collections.emptyIterator()` for the `null` case here.

Reply:
2840f74